src/kbmod_wf/task_impls/reproject_wu.py

```diff
@@ -12,3 +23,289 @@ def reproject_wu(input_wu=None, reprojected_wu=None, logger=None):
+
+        self.pixel_scale = self.uri_params["pixel_scale"]
+        self.patch_size = self.uri_params["patch_size"]
+
+        # handle image dimensions
+        if self.image_width == None or self.image_height == None:
+            if "patch_size" not in self.uri_params:
+                raise KeyError(
+                    f"Must supply image dimensions (image_width, image_height) or #patch_size= must be in a specified URI file."
+                )
+            if self.pixel_scale == None:
+                raise KeyError(
+                    f"When patch pixel dimensions are not specifified, the user must supply a pixel scale via the command line or the uri file."
+                )
+
+        self.image_width, self.image_height = self._patch_arcmin_to_pixels(
+            patch_size_arcmin=self.patch_size,
+            pixel_scale_arcsec_per_pix=self.pixel_scale,
+        )
+
+        self.point_on_earth = EarthLocation(1814303.74553723, -5214365.7436216, -3187340.56598756, unit="m")
+
+    def reproject_workunit(self):
+        self.logger.debug(f"Creating WCS from patch")
+
+        # Create a WCS object for the patch. This will be our common reprojection WCS
+        patch_fits_name = f"patch.fits"
+        if "patch_id" in self.uri_params:
+            patch_fits_name = f"patch_{self.uri_params['patch_id']}.fits"
+
+        # ! TODO: Need to figure out what to do with args.result_dir here. Pass `filename` as an parsl.output?
+        # ! Also, the _create_wcs_from_corners function doesn't make use of all the self.<variables> available to it.
+        patch_wcs = self._create_wcs_from_corners(
+            self.patch_corners,
+            self.image_width,
+            self.image_height,
+            pixel_scale=self.pixel_scale,
+            filename=os.path.join(args.result_dir, patch_fits_name),
+        )
+
+        self.logger.info(f"Reading existing WorkUnit from disk: {self.original_wu_filepath}")
+        orig_wu = WorkUnit.from_fits(self.original_wu_filepath)
+        elapsed = round(time.time() - last_time, 1)
+        self.logger.debug(f"Required {elapsed}[s] to read original WorkUnit {self.original_wu_filepath}.")
+
+        # gather elements needed for reproject phase
+        imgs = orig_wu.im_stack
+
+        # Find the EBD (estimated barycentric distance) WCS for each image
+        last_time = time.time()
+        ebd_per_image_wcs, geocentric_dists = transform_wcses_to_ebd(
+            orig_wu._per_image_wcs,
+            imgs.get_single_image(0).get_width(),
+            imgs.get_single_image(0).get_height(),
+            self.guess_dist,
+            [astropy.time.Time(img.get_obstime(), format="mjd") for img in imgs.get_images()],
+            self.point_on_earth,
+            npoints=10,
+            seed=None,
+        )
+        elapsed = round(time.time() - last_time, 1)
+        self.logger.debug(f"Required {elapsed}[s] to transform WCS objects to EBD..")
```

Comment:
@wilsonbb Generally speaking, there are many of these logger messages that report on timing for a particular step in the processing. I've tried to mark them at `debug` level messages, but I've probably missed a bunch. Feel free to flag or update if you notice them.

Secondary question, how important do you think logging the timing is? We could require a specific format for these messages to make them easier to parse if it's particularly valuable. (I might be inclined to do that regardless of the current perceived value).

Reply:
Yeah a specific format might be helpful here. Something like

f"[Phase {phase name} ({elapsed time in seconds})s] {optional additional msg}"

Also tangentially throwing my personal biases here, I think it's good for all log messages for systems like these to have timestamps. Which I believe we could accomplish with something like

'''
logging.basicConfig(format='%(asctime)s %(message)s')
'''

If not already enabled. But just an optional suggestion since we can also reconstruct elapsed times from our logs that way

Reply:
I'm in full agreement on timestamping each message - I have a default format specified in in the logger_utilities.py module here: https://github.com/dirac-institute/kbmod-wf/blob/main/src/kbmod_wf/utilities/logger_utilities.py#L1-L5

This matches what parsl uses as it's default for logging, so this should make parsing logs much more straightforward.